lib/app/sequence_base.rb

```diff
@@ -841,9 +841,9 @@ def date_comparator_value(comparator, date)
 
         case comparator
         when 'lt', 'le'
-          comparator + (DateTime.xmlschema(date) + 1).xmlschema
+          comparator + (DateTime.xmlschema(date) + 2).xmlschema
```

Comment:
Why adding 2?

Reply:
@czh-orz just explained this to me so let me see if I can explain it right...

Search Parameters for searches are first calculated and a search is performed with them:

e.g.
https://github.com/onc-healthit/inferno-program/blob/cda7ed0acf3a2de8ec2799bca08903964374f54f/lib/modules/uscore_v3.1.0/us_core_procedure_sequence.rb#L222-L225

Then the search parameters for the comparator search are calculated _based on the previous search parameter value_
e.g.
https://github.com/onc-healthit/inferno-program/blob/cda7ed0acf3a2de8ec2799bca08903964374f54f/lib/modules/uscore_v3.1.0/us_core_procedure_sequence.rb#L238

When we do the initial search for a Period `get_value_for_search_param` decrements the date because a start exists
https://github.com/onc-healthit/inferno-program/blob/cda7ed0acf3a2de8ec2799bca08903964374f54f/lib/app/sequence_base.rb#L800-L802

Then as we loop through each of the comparators based on this date any of the `lt` or `le` options will increment the date and put us right back where we started!

https://github.com/onc-healthit/inferno-program/blob/cda7ed0acf3a2de8ec2799bca08903964374f54f/lib/app/sequence_base.rb#L844

So by adding or subtracting 2 instead of one it actually increments or decrements regardless of a previous increment or decrement.

Reply:
@czh-orz is working on a change now that will pull the date from the resource directly in `date_comparator_value` so that it doesn't depend on a previous search parameter calculation

Reply:
Thanks for the explanation Reece. Yeah, I just updated it so that the comparator search will pull from the original resource instead of from the search params.

Reply:
No problem, thank you for explaining it to me originally (and actually making this stuff work)!